components/builder-depot/src/backend/s3.rs

```diff
@@ -0,0 +1,189 @@
+// Copyright (c) 2016-2017 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Pkg storage backend variant which uses S3 (or an API compatible clone) for
+//! hart storage.
+//!
+//! Has been tested against AWS S3.
+//!
+//! All packages are stored in a single bucket, using the fully qualified
+//! package ident as the key, prefixed with `pkgs`
```

Comment:
Why is the "pkgs" key prefix needed / desired?

Reply:
This is actually an incorrect comment, we don't store in a path prefixed with `pkgs`, will get this removed.